caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -195,6 +195,11 @@ def run(
             GeneratedTextResult
                 Generated text result produced by PEFT / Transformers.
         """
+        if device is not None:
+            log.warning(
+                "Specifying device is deprecated and ignored, please update your calling argument"
+            )
+        device = self._DETECT_DEVICE
```

Comment:
I think instead of this, we should fetch the device information from the `model` itself, otherwise there is still possibility of discrepancy. Can we instead do `self.model.device` to fetch the information which device the model is ?

Reply:
The old line 177 set device this way, I assumed there was a reason for it?

Reply:
Hmm. in grand scheme of things, it was sort of a bug 😄

Reply:
Yeah, it actually is a bug since the device has to match the model, otherwise it'll throw because the tensors are on wrong devices. Probably the right behavior here is to remove this parameter from `.run` (since we are in `0.x` still) rather than keep it here, since the only way this parameter can currently be used should be just putting it on the model device